### DISCUSSION TITLE
找不到远程工具链

### DISCUSSION THREAD

**Discussion by HelloWorld886** [xmake-io/xmake#2931] [Q&A]
### Xmake 版本

2.6.4

### 操作系统版本和架构

Windows 10

### 描述问题

自定义仓库远程工具链，使用时the toolchain not found。

### 期待的结果

能正常使用远程工具链

### 工程配置

add_repositories("xswitch-repo https://github.com/HelloEngine/xswitch-repo.git main")
add_requires("devkit-a64")
target("application")
set_toolchains("@devkit-a64")

### 附加信息和错误日志

error: @programdir\core\main.lua:280: @programdir\core\project\target.lua:1956: the toolchain devkit-a64 not found!
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:849]:
    [@programdir\core\project\target.lua:1956]: in function 'toolchains'
    [@programdir\actions\config\main.lua:136]: in function '_check_target_toolchains'
    [@programdir\actions\config\main.lua:427]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:280]:
    [@programdir\core\base\task.lua:519]: in function 'run'
    [@programdir\core\main.lua:278]: in function 'cotask'
    [@programdir\core\base\scheduler.lua:371]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:849: in function 'os.raiselevel'
        (...tail calls...)
        @programdir\core\main.lua:280: in upvalue 'cotask'

**Comment by waruqi** [comment]
使用指定的工具链 foo 并绑定包 bar。

```lua
set_toolchains("foo@bar")
```

使用指定的工具链 bar 并绑定包 bar。

```lua
set_toolchains("@bar")
```

> set_toolchains("@devkit-a64")

你这没有定义 `toolchain("devkit-a64")` ，也没指定已知的内置 toolchain，只有一个包，除了拉取下载，啥也没用。。

参考 https://github.com/xmake-io/xmake/blob/554ab8a3cdb805efc83870d22d01790df4e55782/tests/projects/package/toolchain_muslcc/xmake.lua#L36

它绑定到了自定义的 muslcc 工具链。
